server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DatabasePermissionController.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.server.web.controller.v2;
+
+import java.util.List;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.oceanbase.odc.core.shared.constant.AuthorizationType;
+import com.oceanbase.odc.service.common.response.ListResponse;
+import com.oceanbase.odc.service.common.response.PaginatedResponse;
+import com.oceanbase.odc.service.common.response.Responses;
+import com.oceanbase.odc.service.common.response.SuccessResponse;
+import com.oceanbase.odc.service.permission.database.DatabasePermissionService;
+import com.oceanbase.odc.service.permission.database.model.CreateDatabasePermissionReq;
+import com.oceanbase.odc.service.permission.database.model.DatabasePermissionType;
+import com.oceanbase.odc.service.permission.database.model.QueryDatabasePermissionParams;
+import com.oceanbase.odc.service.permission.database.model.UserDatabasePermission;
+
+import io.swagger.annotations.ApiOperation;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/1/4 10:15
+ */
+@RestController
+@RequestMapping("api/v2/collaboration/projects/{projectId:[\\d]+}/databasePermissions")
```

Comment:
better add `/api/v2/collaboration/projects/{projectId:[\\d]+}/databasePermissions` either.

Reply:
You are right.